Setup Totp model and factories

jaspervanbrian opened the pull request:
- Install byebug for debugging
- Create migration for totp
- Fix 'Factory not defined'
- Add stellar-sdk dependency
- Default to rbnacl 5.0.0 (v6.0.0 doesn't support rbnacl-libsodium)
- Require factory_bot_rails instead of factory_bot

ramontayag commented on app/controllers/stellar_multisig/rails/api/v1/totp_controller.rb:
```diff
@@ -0,0 +1,6 @@
+require_dependency "stellar_multisig/rails/application_controller"
+
+module StellarMultisig::Rails
```
@jaspervanbrian it looks having `::Rails` for a Rails engine [is recommended](https://stackoverflow.com/a/8961409/61018), but it bothers me that the migrations have a `rails` in the table name.

ramontayag replied:
I think what will work is to [set the engine name](https://edgeapi.rubyonrails.org/classes/Rails/Engine.html) like:

```ruby
module MyEngine
  class Engine < Rails::Engine
    engine_name "my_engine"
  end
end
```

Then, when you create migrations, `rails` should not be there anymore.

ramontayag commented on stellar_multisig-rails.gemspec:
```diff
@@ -17,6 +17,8 @@ Gem::Specification.new do |s|
   s.files = Dir["{app,config,db,lib}/**/*", "MIT-LICENSE", "Rakefile", "README.md"]
 
   s.add_dependency "rails", "~> 5.2.1"
+  s.add_dependency "rbnacl", "~> 5.0.0"
```
```suggestion
  s.add_dependency "rbnacl", "< 6.0"
```

jaspervanbrian commented:
I removed `::Rails` from `isolate_namespace`, because when you generate something like a model or a controller, it auto nests it with the namespace you provided in the `isolate_namespace`.

I did this because it is easier to generate controller with a `Rails` namespace by just `rails g controller Rails::<controller_path>` command, than to avoid a migration with a `_rails_` name on it.

I also added the engine_name "/stellar_multisig", because it looks much more nicer than "/stellar_multisig-rails"

jaspervanbrian commented:
Actual format of the totp.provisioning_uri:
```
otpauth://totp/<issuer>:<string_provided>?secret=<totp.secret>&issuer=<totp.issuer>
```

```
totp = ROTP::TOTP.new("asd", issuer: "qwe")
totp.provisioning_uri("alice@google.com") 
# => 'otpauth://totp/qwe:alice@google.com?secret=asd&issuer=qwe'
```
